src/middlewared/middlewared/etc_files/nfsd.py

```diff
@@ -88,6 +90,24 @@ def build_share_targets(share):
     except Exception:
         pass
 
+    # Workaround to work with "onetime", since the rc scripts depend on rc flags.
```

Comment:
Not sure what I feel about mixing config file generation with calling service start/stop.

Also NFS is picky about services order and how its done at boot (which we will be doing with this change).

I imagine its now broken because we do reload and these are not called? How about make it a common method in service.py and also do it on reload?

Reply:
> I imagine its now broken because we do reload and these are not called?

We call `etc.generate nfsd` on reload, so this is called

Reply:
> Not sure what I feel about mixing config file generation with calling service start/stop.

We already do this in other files, that's why we have a design proposal to fix at least some of this in 12

Reply:
Yes, i know this is called on reload but it werent before, we did a full restart. So why not just call these on service reload instead too? I worry about regressions on startup.

Also, which other etc plugins we are calling service?

Reply:
> I worry about regressions on startup.

The ticket actually fixes this regression. I am struggling to find what was calling `service.start nfs` on startup when it worked..

Reply:
service.start is not called on startup at all. Its plain and simple rc.d with rc.conf.local

Reply:
So what was setting this sysctl then?

Reply:
rc.conf.local

Reply:
Or it was supposed to, and that may be the real issue. Its where other sysctls are supposed to be set in 11.2

Reply:
Taking a deeper look: /etc/rc.d/nfsd is supposed to set it on boot

Reply:
Changing this particular parameter doesn't actually require a reload or restart of the service. Just FYI

Reply:
We are not talking about full services reload/restart, speaking on terms of our shim in the service layer

Reply:
Yes, and `vfs.nfsd.enable_stringtouid` is set by `rc.conf`, so the only problem left is that neither `/etc/rc.d/nfsd` nor `rc.conf` are not called on `service.reload nfs`, which is fixed now.